First version of site

RizoBoy opened the pull request.

RizoBoy committed:
Init commit

RizoBoy committed:
added link to site in README

stepancar commented on contact.html:
```diff
@@ -0,0 +1,109 @@
+                                </div>
+                                <div class="col-12">
+                                    <label for="email" class="formLabel">E-mail (опционально)</label>
+                                    <input type="text" class="form-control" id="email" placeholder="address@example.com">
```
давайте добавим валидацию для почты

stepancar commented on js/Toast.min.js:
```diff
@@ -0,0 +1 @@
+"use strict";function Toast(t){if(!t.message)throw new Error("Toast.js - You need to set a message to display");this.options=t,this.options.type=t.type||"default",this.toastContainerEl=document.querySelector(".toastjs-container"),this.toastEl=document.querySelector(".toastjs"),this._init()}Toast.prototype._createElements=function(){var t=this;return new Promise(function(e,o){t.toastContainerEl=do ...
```
давайте найдем этот пакет в cdn

stepancar commented on js/index.js:
```diff
@@ -0,0 +1,93 @@
+
+(function() {
+    $.fn.isInViewport = function(element) {
+        var distance = element.getBoundingClientRect();
```
в коде ниже вы используете const. Пожалуйста, добейтесь консистентности в коде

stepancar commented on js/index.js:
```diff
@@ -0,0 +1,93 @@
+    if (!triggeredCounters) {
+        jQuery(function($) {
+            document.querySelectorAll('.counter').forEach((element) => {
+                if ($().isInViewport(element)) {
```
Пока не очень понял идею. Если вы пишите именно jquery plugin то предполагается что элементы в него  передаются как это обычно делается в jquery: $(element).isInViewport(). 
Или идея в другом?

stepancar commented on js/index.js:
```diff
@@ -0,0 +1,93 @@
+                    const classes = element.className.split(' ');
+                    classes[classes.length - 1] += '-counter';
+    
+                    element.className = classes.join(' ');
+                        $('.years-counter').countTo({
+                            from: 0,
+                            to: 5,
+                            speed: 2000,
+                            refreshInterval: 20
+                        });
+                        $('.members-counter').countTo({
```
пожалуйста, отформатируйте код перед отправкой на ревью

stepancar reviewed (changes requested):
Оствил комментарии

RizoBoy commented:
@stepancar выполнил правки, проверьте пожалуйста.

RizoBoy committed:
PR fixes

1. Toast moved to cdn.
2. Added validation for email.
3. Added ensure consistency.
4. Formated some code.
5. Moved the same parameter to another bracket in counters.

stepancar commented on js/index.js:
```diff
@@ -0,0 +1,94 @@
+        jQuery(function($) {
+            document.querySelectorAll('.counter').forEach((element) => {
+                if ($(element).isInViewport()) {
+                    const classes = element.className.split(' ');
```
в дом есть возможность получить классы отдельно

stepancar commented on js/index.js:
```diff
@@ -0,0 +1,94 @@
+            document.querySelectorAll('.counter').forEach((element) => {
+                if ($(element).isInViewport()) {
+                    const classes = element.className.split(' ');
+                    classes[classes.length - 1] += '-counter';
```
а что если этот клаасс добавлен не в конец? Давайте сделаем надежно!

stepancar commented on index.html:
```diff
@@ -0,0 +1,148 @@
+                    <div class="col-lg-6">
+                        <h1>Ризобоев Андрей</h1>
+                        <p class="lead">
+                            Студент ИТМО третьего курс по направлению “Нейротехнологии и программирование”. Более 5 лет занимаюсь открытием и развитием игровых серверов в качестве полноценных прибыльных проектов, благодаря чему хорошо обладаю рассудительными и организационными навыками. Создал ИИ психолога с помощью машинного обучения в качестве телеграмм бота. Всегда готов присоединиться к коман ...
```
```suggestion
                            Студент ИТМО третьего курса по направлению “Нейротехнологии и программирование”. Более 5 лет занимаюсь открытием и развитием игровых серверов в качестве полноценных прибыльных проектов, благодаря чему хорошо обладаю рассудительными и организационными навыками. Создал ИИ психолога с помощью машинного обучения в качестве телеграмм бота. Всегда готов присоединиться к команде для решения задач.
```

stepancar commented on index.html:
```diff
@@ -0,0 +1,148 @@
+                        <div>
+                            <h6 class="text-sm">Обратная связь</h6>
+                            <ul class="flex-column ms-n3 nav">
+                                <li class="nav-item">E-mail: kyles.scpsl@gmail.com</li>
```
Давайте сделаем ссылку таким образом чтобы при нажатии на нее открывался нативный клиент

stepancar commented on index.html:
```diff
@@ -0,0 +1,148 @@
+                            <h6 class="text-sm">Обратная связь</h6>
+                            <ul class="flex-column ms-n3 nav">
+                                <li class="nav-item">E-mail: kyles.scpsl@gmail.com</li>
+                                <li class="nav-item">Discord: rizoboy</li>
```
Давайте добавим ссылку

stepancar reviewed (changes requested):
На защиту давайте сделаем аналитику вашей страницы

RizoBoy committed:
Update index.html

Co-authored-by: Stepan Mikhailiuk <istepancar@gmail.com>

RizoBoy commented:
@stepancar снова выполнил правки, проверьте пожалуйста.

RizoBoy committed:
PR fixes 2

1. added links to mail and discord
2. changed webhook
3. formatted code
4. slightly reworked counters

stepancar commented:
@RizoBoy , new увидел код относящийся к https://github.com/stepancar-web-programming/2023-fall-lab-portfolio-rizoboev-a./pull/1#pullrequestreview-1831739043

RizoBoy committed:
add google analytics

stepancar commented:
Поскольку ПР был создан почти через 1.5 месяца после дедлайна, ставлю минимальный балл.
